src/main/java/ca/corefacility/bioinformatics/irida/service/impl/RemoteAPITokenServiceImpl.java

```diff
@@ -99,83 +100,90 @@ public RemoteAPIToken updateTokenFromRefreshToken(RemoteAPI api) {
 		try {
 			token = getToken(api);
 
-			String refreshToken = token.getRefreshToken();
+			String refreshTokenValue = token.getRefreshToken();
+
+			if (refreshTokenValue != null) {
+				RefreshToken refreshToken = new RefreshToken(refreshTokenValue);
 
-			if (refreshToken != null) {
 				URI serviceTokenLocation = UriBuilder.fromUri(api.getServiceURI()).path("oauth").path("token").build();
 
-				OAuthClientRequest tokenRequest = OAuthClientRequest.tokenLocation(serviceTokenLocation.toString())
-						.setClientId(api.getClientId())
-						.setClientSecret(api.getClientSecret())
-						.setRefreshToken(refreshToken)
-						.setGrantType(GrantType.REFRESH_TOKEN)
-						.buildBodyMessage();
+				ClientAuthentication clientAuth = new ClientSecretBasic(new ClientID(api.getClientId()),
+						new Secret(api.getClientSecret()));
+
+				TokenRequest tokenRequest = new TokenRequest(serviceTokenLocation, clientAuth,
+						new RefreshTokenGrant(refreshToken));
+
+				TokenResponse tokenResponse = TokenResponse.parse(tokenRequest.toHTTPRequest().send());
+
+				if (!tokenResponse.indicatesSuccess()) {
+					// We got an error response...
+					TokenErrorResponse errorResponse = tokenResponse.toErrorResponse();
+					logger.error("Updating token by refresh token failed", errorResponse.getErrorObject().toString());
+				} else {
+
+					AccessTokenResponse accessTokenResponse = tokenResponse.toSuccessResponse();
 
-				OAuthJSONAccessTokenResponse accessToken = oauthClient.accessToken(tokenRequest);
+					token = buildTokenFromResponse(accessTokenResponse, api);
 
-				token = buildTokenFromResponse(accessToken, api);
+					delete(api);
+					token = create(token);
 
-				delete(api);
-				token = create(token);
+					logger.trace("Token for api " + api + " updated by refresh token.");
 
-				logger.trace("Token for api " + api + " updated by refresh token.");
+				}
 			} else {
 				logger.trace("No refresh token for api " + api + ". Cannot update access token.");
 			}
 		} catch (EntityNotFoundException ex) {
 			logger.debug("Token not found for api " + api + ".  Cannot update access token.");
-		} catch (OAuthProblemException | OAuthSystemException ex) {
-			logger.error("Updating token by refresh token failed", ex.getMessage());
+		} catch (ParseException e) {
+			// TODO Auto-generated catch block
+			e.printStackTrace();
```

Comment:
I think these should be `logger.error("Updating token by refresh token failed", e)` so that the exception and stack trace winds up in our log files.

Reply:
Refactored in 9aad969ccf